lib/gui/components/safe-webview/components/safe-webview.js

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const react = require('react');
+const propTypes = require('prop-types');
+
+/**
+ * @summary Webviews that time out
+ * @type {Object}
+ * @public
+ */
+class SafeWebview extends react.PureComponent {
+
+  /**
+   * @param {Object} props - React element properties
+   */
+  constructor(props) {
+    super(props);
+
+    /**
+     * @summary Loading timeout ID
+     * @type Number
+     * @public
+     */
+    this.loadingTimeoutId = null;
+    this.state = {
+      shouldLoad: true
+    };
+
+    // Events steal 'this'
+    this.didStartLoading = this.didStartLoading.bind(this);
+    this.didFailLoad = this.didFailLoad.bind(this);
+    this.didStopLoading = this.didStopLoading.bind(this);
+    this.didGetResponseDetails = this.didGetResponseDetails.bind(this);
+  }
+
+  /**
+   * @returns {react.Element}
+   */
+  render() {
+    if (this.state.shouldLoad) {
+      return react.createElement('webview', {
+        ref: 'webview',
+        src: this.props.src,
+        style: {
+
+          // Hide/show the webview per the documented way
+          // https://electron.atom.io/docs/api/webview-tag/#css-styling-notes
+          flex: this.state.shouldLoad ? null : '0 1',
+          width: this.state.shouldLoad ? null : '0',
+          height: this.state.shouldLoad ? null : '0'
+        }
+      }, []);
+
+    }
+
+    return null;
```

Comment:
Just FYI we have to explicitly return `null` here because `undefined` is an error in React.